### ISSUE TITLE
Can't capture vulkan on ubuntu machine

### ISSUE THREAD

**Issue by kenchen3000** [baldurk/renderdoc#927]
## Description
I am trying to capture vulkan application on ubuntu but can't find way to capture it. I am running qrenderdoc and it loads the app correctly. but F12 or print doesn't work, and i don't see any menu button for capture.(actually the trigger after delay button is greyed out)
It also prompts vulkan layer is not registered, so i follow the option to register. but the next time i open renderdoc, it still mentions vulkan layer is not registered. I don't know if it relates to the problem.

thanks.

## Repro steps
Launch renderdoc with vulkan app, press F12, or print screen, nothing happens.

## Environment

* RenderDoc build: v1.0 downloaded.
* Operating System: Ubuntu Linux
* API:  Vulkan

**Comment by kenchen3000** [comment]
I think i figured out somehow. I had to manually run sudo ./qrenderdoc --install_vulkan_layer 1
(i don't even know what 1 means...)
it seems the script run by qrenderdoc interface to register vulkan layer was stuck for unknown reason.

**Comment by baldurk** [comment]
If you're willing to help investigate this I'd like to know why the vulkan layer registration was failing. The `1` in this case is just a flag meaning it's a system-level install rather than user-level.

It sounds like you clicked yes to install at system level rather than user level. If you remove the layer registration again by hand, can you successfully install at user level that doesn't do a `sudo` ?

Then my guess is something went wrong trying to launch a graphical `sudo`. RenderDoc tries `pkexec`, `kdesudo`, `gksudo`, `beesu` in that order - do you have any of them installed? If not, then it tries to launch `x-terminal-emulator`, `gnome-terminal`, `knosole` (oops, I only just noticed this typo) and `xterm` in that order, to run `sudo` manually.

If you could tell me which of these you have available and which renderdoc tries to use, that would be helpful. Then maybe we can track down why it got stuck.

**Comment by kenchen3000** [comment]
i have pkexec, gksudo.
i am not sure how to remove layer registration. but i happened to install it on another ubuntu machine and had some issue. running without sudo works as well:  ./qrenderdoc --install_vulkan_layer 1
hope it helps.

**Comment by krupitskas** [comment]
I can confirm everything @kenchen3000 said above. Also solved by this "system install command with 1"

**Comment by baldurk** [comment]
Looking at this more closely, the parameter to `--install_vulkan_layer` should either be `root` or `user` indicating whether to install at system or user level. Where did you get the `1` parameter from?

In this case anything that doesn't match `root` will be treated as `user`, so the layer will have been registered in your home directory.

To investigate further:

1. Removing the layer registration will most likely just mean removing either `$XDG_DATA_HOME/vulkan/implicit_layer.d/renderdoc_capture.json` if `$XDG_DATA_HOME` is set, or else `$HOME/.local/share/vulkan/implicit_layer.d/renderdoc_capture.json` if not.
2. Now you should be back to the initial state. This time, instead of choosing to install the layer at system level with root privileges, select to install it as a user. **This should work** as it's equivalent to the manual command you ran. I just want to check that it does actually work through the normal workflow.
3. Remove the layer registration again, and this time try manually running the command to register as root: `pkexec /path/to/bin/qrenderdoc --install_vulkan_layer root`. See if you get an error message or something from this. If this succeeds you should end up with a `/etc/vulkan/implicit_layer.d/renderdoc_capture.json` file registered. If this doesn't work from the normal UI workflow something must be going wrong with pkexec.

Let me know what you find, hopefully there will be an obvious error when you run `pkexec` manually.

**Comment by kenchen3000** [comment]
I can confirm install it as a user in step 2 doesn't work. it is same as install as system level.
For using 1 as parameter of ./qrenderdoc --install_vulkan_layer, i just pick it randomly because it needs a parameter, luckily it just work...

**Comment by baldurk** [comment]
That seems very strange, it should be running the same command. If you run `./qrenderdoc --install_vulkan_layer user` by hand does that work? Can you check to see if there's any difference in permissions or environment when running the UI normally (and then letting it execute that command) vs you running it from the command line?

**Comment by baldurk** [comment]
Just to bump this one last time, there's clearly an issue here where either the graphical sudo commands or even just launching processes as a user isn't functioning as expected, but I need someone to help me debug exactly what is going on.

**Comment by baldurk** [comment]
Without a repro case or someone who can help me debug this with more information, I can't investigate this further. If you run into this issue, please get in touch either by commenting here or opening a new issue.

**Comment by ShabbyX** [comment]
I have the same issue, in that every time I open renderdoc, it asks to register the vulkan layer again. I asked it to do a system level install.

I then went to copy the layer to /etc, which made renderdoc complain (additionally) that there is another (conflicting) registration that should be removed.

Following this thread, I removed the copy I made and asked renderdoc to register the layer as a user. This made the warning go away.

If you are interested in debugging the issue with the graphical sudo command, I could help you with testing.
